include/meta/meta.hpp

```diff
@@ -2125,6 +2125,21 @@ namespace meta
             using min = defer<min, T, U>;
         }
 
+        /// Index of the first occurrence of the type \p T within the list \p List.
+        /// \par Complexity
+        /// \f$ O(N) \f$.
+        /// \ingroup list
+        template <typename T, typename List>
```

Comment:
I thought about it, originally I had it implemented like:

```c++
template<typename T, typename List>
using index = if_c<List::size() - find<List, T>::size() == 0, 
                   meta::nil_, 
                   List::size() - find<List, T>::size()>;
```

because you can pattern match on `meta::nil_` which is nice, and pattern matching on zero might mean either that it is the first element, or that it is not found if the list has zero size. 

But I liked this more because I found it to be simpler.